src/workloads/execution/SecondaryReadsGenny.yml

```diff
@@ -0,0 +1,252 @@
+            s: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+          Indexes:
+          - keys: {x: 1}
+
+- TemplateName: QuiesceTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: Quiesce}}
+    Type: QuiesceActor
+    Threads: 1
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Repeat: 1
+            
+
+- TemplateName: WritesOnPrimaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: WritesOnPrimary}}
+    Type: CrudActor
+    Threads: *writer_threads
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *test_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: insertOne
+            OperationCommand:
+              Document:
+                x: {^RandomInt: {min: *initial_nb_docs, max: 1000000000}}
+                #s: {^Repeat: {count: *document_size, fromGenerator: {^Choose: {from: ["x"]}}}} --> why does this not work?
+                s: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+- TemplateName: ReadsOnSecondaryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: ReadsOnSecondary}}
+    Type: CrudActor
+    Threads: {^Parameter: {Name: "Threads", Default: -1}}
+    ClientName: Secondary
+    Database: *db
+    Phases:
+      OnlyActiveInPhases:
+        Active: {^Parameter: {Name: "ActivePhases", Default: [-1]}}
+        NopInPhasesUpTo: *max_phase
+        PhaseConfig:
+          Duration: *test_duration
+          Collection: Collection0
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: { x: {$gte: {^RandomInt: {min: 0, max: *initial_nb_docs}}}} # this should be *initial_nb_docs - *read_limit but we don't seem to have and add/subtract generator and it doesn't really impact performance
```

Comment:
Correct. We don't have shared state here. We could make it larger, but it might return nothing some of the time. 
The question in my mind is if wt cache effects matter, and I suspect he answer is no for this workload, but maybe the replication team could chime in here as well.

Reply:
The original test ensures that the reads never go outside the initial documents:

https://github.com/10gen/workloads/blob/083f9d81c593c07a69d762b84b06dc4b55b628e5/workloads/secondary_reads.js#L160

I wasn't able to do the same thing dynamically using genny as I couldn't find any way to add/subtract two values via a generator. Maybe the simples workaround here would be to add the result of the subtraction as a static parameter at the top?